dll/win32/framedyn/wbemtime.cpp

```diff
@@ -0,0 +1,670 @@
+/*
+* COPYRIGHT:       See COPYING in the top level directory
+* PROJECT:         ReactOS system libraries
+* FILE:            dll/win32/framedyn/wbemtime.cpp
+* PURPOSE:         WBEMTime class implementation
+* PROGRAMMERS:     Jin Juanshi (2033537949@qq.com)
+*/
+
+#include <wbemtime.h>
+#include <stdio.h>
+#include <time.h>
+#include <strsafe.h>
+
+#define SECONDS_PER_MINUTE 60
+#define SECONDS_PER_HOUR SECONDS_PER_MINUTE * 60
+#define SECONDS_PER_DAY SECONDS_PER_HOUR * 24
+
+#define MICORSECONDS_OFFSET 10
+#define MILLISECONDS_OFFSET MICORSECONDS_OFFSET * 1000
+#define SECONDS_OFFSET MILLISECONDS_OFFSET * 1000
```

Comment:
When you define macros that perform arithmetic operations like these, you should put parentheses around the operation (here for example: `#define SECONDS_OFFSET (MILLISECONDS_OFFSET * 1000)`) because it may happen that, depending on where and how you use the macro, its expansion would not do what you expect it to do.

Reply:
Thanks.